addons/survey/tests/test_survey_ui_session.py

```diff
@@ -140,70 +141,80 @@ def test_admin_survey_session(self):
-        attendee_2.save_lines(simple_choice_question, simple_choice_answer_1.id)
-        attendee_3.save_lines(simple_choice_question, simple_choice_answer_2.id)
-        attendee_1.save_lines(scored_choice_question, scored_choice_answer_1.id)
-        attendee_2.save_lines(scored_choice_question, scored_choice_answer_2.id)
-        attendee_3.save_lines(scored_choice_question, scored_choice_answer_3.id)
-        attendee_1.save_lines(timed_scored_choice_question,
-            [timed_scored_choice_answer_1.id, timed_scored_choice_answer_3.id])
-        attendee_2.save_lines(timed_scored_choice_question,
-            [timed_scored_choice_answer_1.id, timed_scored_choice_answer_2.id])
-        attendee_3.save_lines(timed_scored_choice_question,
-            [timed_scored_choice_answer_2.id])
-
-        self.start_tour('/web', 'test_survey_session_manage_tour', login='admin')
-
-        self.assertFalse(bool(survey_session.session_state))
-        self.assertTrue(all(answer.state == 'done' for answer in all_attendees))
+        def action_open_session_manager_mock(self):
+            self.ensure_one()
+
+            return {
+                'type': 'ir.actions.act_url',
+                'name': "Open Session Manager",
+                'target': 'self',
+                'url': '/survey/session/manage/%s' % self.access_token
+            }
+
+        with patch('odoo.addons.survey.models.survey_survey.Survey.action_open_session_manager', action_open_session_manager_mock):
+            self.start_tour('/web', 'test_survey_session_create_tour', login='admin')
+
+            # tricky part: we only take into account answers created after the session_start_time
+            # the create_date of the answers we just saved is set to the beginning of the test.
+            # but the session_start_time is set after that.
+            # So we cheat on the session start date to be able to count answers properly.
+            survey_session.write({'session_start_time': test_start_time - relativedelta(minutes=10)})
+
+            attendee_1 = survey_session._create_answer()
+            attendee_2 = survey_session._create_answer()
+            attendee_3 = survey_session._create_answer()
+            all_attendees = [attendee_1, attendee_2, attendee_3]
+
+            self.assertEqual('ready', survey_session.session_state)
+            self.assertTrue(all(attendee.is_session_answer for attendee in all_attendees),
+                "Created answers should be within the session.")
+            self.assertTrue(all(attendee.state == 'new' for attendee in all_attendees),
+                "Created answers should be in the 'new' state.")
+
+            # =========================================
+            # PART 2 : OPEN SESSION AND CHECK ATTENDEES
+            # =========================================
+
+            self.start_tour('/web', 'test_survey_session_start_tour', login='admin')
```

Comment:
Does it not work if you only put this single line within your "with" and leave the rest outstide of it?
Would lessen the diff a lot.